src/app/firebase_app.ts

```diff
@@ -474,36 +491,30 @@ export function createFirebaseNamespace(): FirebaseNamespace {
     if (factories[name]) {
       error('duplicate-service', {'name': name});
     }
-    if (!!allowMultipleInstances) {
-      // Check if the service allows multiple instances per app
-      factories[name] = createService;
-    } else {
-      // If not, always return the same instance when a service is instantiated
-      // with an instanceString different than the default.
-      factories[name] =
-          (app: FirebaseApp, extendApp?: (props: {[prop: string]: any}) => void,
-           instanceString?: string) => {
-            // If a new instance is requested for a service that does not allow
-            // multiple instances, return the default instance
-            return createService(app, extendApp, DEFAULT_ENTRY_NAME);
-          };
-    }
+
+    /**
+     * If multiple instances are allowed, return the true create service
+     * otherwise, return a proxied reference to the same service
+     */
+    factories[name] = allowMultipleInstances ? createService :
+      (...args) => createService.apply(this, args);
```

Comment:
Seems like this ternary statement isn't really doing anything? Before, the if statement was always sending DEFAULT_ENTRY_NAME instead of the passed instanceString if `!allowMultipleInstances`, but since you handle the multiple instance name thing further down now (inside the function assigned to `FirebaseAppImpl.prototype[name]`), you should just be able to do

    factories[name] = createService;
and have it work, right?